### PR TITLE
fix(NcRichContenteditable): adjust to new design

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

- Fix https://github.com/nextcloud-libraries/nextcloud-vue/issues/5975

### 🚧 Tasks

- [x] Use new and correct border-radious variables with old values as a fallback
- [x] Adjust paddings to remove double paddings
  - Paddings adjustments also applies for 28+29 intentionally (previously there were double paddings)
- [x] Add `box-sizing: border-box` so it doesn't break if mounted to body
- [x] Change style of autocomplete using 1.2 line-height to fix cutting issue
  - Also applies for 28 + 29, but not a visible change

Note, it has some copy-paste from `NcAvatar` and `NcListItem`. We can look at reuse in a follow-up

### 🖼️ Screenshots

🏚️ Before 30 | 🏡 After 30
---|---
image | image
image | image
image | image

🏚️ Before 28 | 🏡 After 28
---|---
image | image
image | image
image | image

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

**Comment by ShGKme** [comment]
Requires some more fixes to prevent changes in 28

File: src/components/NcRichContenteditable/NcAutoCompleteResult.vue [hunk 1]
```diff
@@ -127,21 +129,22 @@ export default {
 	}
 
 	&__status {
+		--auto-complete-result-status-icon-size: clamp(14px, var(--auto-complete-result-avatar-size) * 0.4, 18px);
+		// 2 * Avatar Radius * (1 - 1 / sqrt(2)) - Status Icon Size / 2
+		--auto-complete-result-status-icon-position: calc(var(--auto-complete-result-avatar-size) / 2 * (1 - 1 / sqrt(2)) - var(--auto-complete-result-status-icon-size) / 2);
```
Comment (ShGKme) [inline src/components/NcRichContenteditable/NcAutoCompleteResult.vue:134]: image

Comment (ShGKme) [reply src/components/NcRichContenteditable/NcAutoCompleteResult.vue:134]: If it goes well, I'll try to adapt it to NcAvatar and then use NcAvatar here to not duplicate it.

File: src/components/NcRichContenteditable/NcAutoCompleteResult.vue [hunk 2]
```diff
@@ -127,21 +129,22 @@ export default {
 	}
 
 	&__status {
+		--auto-complete-result-status-icon-size: clamp(14px, var(--auto-complete-result-avatar-size) * 0.4, 18px);
+		// 2 * Avatar Radius * (1 - 1 / sqrt(2)) - Status Icon Size / 2
```
Comment (Antreesy) [inline src/components/NcRichContenteditable/NcAutoCompleteResult.vue:133]:
```suggestion
		// Avatar Radius * (1 - 1 / sqrt(2)) - Status Icon Size / 2
```

**Review by Antreesy — APPROVED** [review]
Tested with Talk

File: src/components/NcRichContenteditable/NcAutoCompleteResult.vue (hunk 2 above)
Comment (ShGKme) [reply src/components/NcRichContenteditable/NcAutoCompleteResult.vue]: Done

File: src/components/NcRichContenteditable/NcAutoCompleteResult.vue [hunk 3]
```diff
@@ -127,21 +129,22 @@ export default {
 	}
 
 	&__status {
+		--auto-complete-result-status-icon-size: clamp(14px, var(--auto-complete-result-avatar-size) * 0.4, 18px);
+		// Avatar Radius * (1 - 1 / sqrt(2)) - Status Icon Radius / 2
```
Comment (susnux) [inline src/components/NcRichContenteditable/NcAutoCompleteResult.vue:133]: Fancy

File: src/components/NcRichContenteditable/NcAutoCompleteResult.vue [hunk 4]
```diff
@@ -127,21 +129,22 @@ export default {
...
+		--auto-complete-result-status-icon-position: calc(var(--auto-complete-result-avatar-size) / 2 * (1 - 1 / sqrt(2)) - var(--auto-complete-result-status-icon-size) / 2);
 		box-sizing: border-box;
 		position: absolute;
-		right: -4px;
-		bottom: -4px;
-		min-width: 18px;
-		min-height: 18px;
-		width: 18px;
-		height: 18px;
+		right: var(--auto-complete-result-status-icon-position);
```
Comment (susnux) [inline src/components/NcRichContenteditable/NcAutoCompleteResult.vue:137]:
```suggestion
		inset-inline-end var(--auto-complete-result-status-icon-position);
```

File: src/components/NcRichContenteditable/NcRichContenteditable.vue [hunk 5]
```diff
@@ -1246,7 +1247,7 @@ export default {
 			width: 20px;
 			height: 20px;
 			object-fit: contain;
-			padding-right: calc(var(--default-grid-baseline) * 2);
+			padding-right: var(--default-grid-baseline);
```
Comment (susnux) [inline src/components/NcRichContenteditable/NcRichContenteditable.vue:1250]:
```suggestion
			padding-inline-end: var(--default-grid-baseline);
```

**Review by susnux — APPROVED** [review]
Nice, just some comments to make like easier in the NC31 cycle

**Pull request merged**